web/app.js

```diff
@@ -802,7 +802,15 @@ const PDFViewerApplication = {
     }
 
     if (!this.pdfLoadingTask) {
-      return undefined;
+      return;
+    }
+    if (
+      (typeof PDFJSDev === "undefined" || PDFJSDev.test("GENERIC")) &&
+      this.pdfDocument?.annotationStorage.size > 0 &&
```

Comment:
Technically the following check *should* be sufficient here, but this line cannot hurt and should help avoid triggering saving unnecessarily.